clang/lib/Driver/ToolChains/SYCL.h

```diff
@@ -20,20 +20,18 @@ class Command;
 namespace tools {
 namespace SYCL {
 
-// Gather command line arguments for backend compilation call
-void TranslateSYCLTargetArgs(Compilation &C,
-              const llvm::opt::ArgList &Args, const ToolChain &TC, llvm::opt::ArgStringList &CmdArgs);
-
-// Gather command line arguments for device specific link
-void TranslateSYCLLinkerArgs(Compilation &C,
-              const llvm::opt::ArgList &Args, const ToolChain &TC, llvm::opt::ArgStringList &CmdArgs);
-
 void constructLLVMForeachCommand(Compilation &C, const JobAction &JA,
                                  std::unique_ptr<Command> InputCommand,
                                  const InputInfoList &InputFiles,
                                  const InputInfo &Output, const Tool *T,
                                  StringRef Ext);
 
+void TranslateBackendTargetArgs(const ToolChain &TC,
```

Comment:
Could `this` be used instead of `TC`? I understand that SYCL BE compilers' `ConstructJob()` methods can rely on the fact that it's a `SYCLToolChain`.

Reply:
Given the current usage `this` cannot be used instead of TC.  The translations are just workers against existing command lines and are not new ToolChain objects like the BE compilers.

Reply:
My miss these weren't declared as `SYCLToolChain` members. Could this be done, though?

Reply:
Yup - done.